src/main/java/Player.java

```diff
@@ -13,8 +13,25 @@ public Player (String name, char color) {
     }
 
     public Player copy() {
-        // TODO
-        return null;
+        //PlayerCopier pc = new PlayerCopier(this);
+        //return pc.createCopy();
+        String name = this.name;
+        char color = this.color;
+        Player newPlayer = new Player(name, color);
+        
+        // Points will autmatically be calculated when captured pieces are added
+        List<Piece> capturedPieces = this.oldPlayer.getCapturedPieces();
+        for (Piece piece : capturedPieces) {
+            Piece newPiece = piece.copy();
+            newPlayer.addCapturedPiece(newPiece);
+        }
+
+        List<Piece> pieceList = this.oldPlayer.getPieceList();
+        for (Piece piece : pieceList) {
+            Piece newPiece = piece.copy();
+            newPlayer.addPiece(newPiece);
```

Comment:
Am I mistaken or do `Player` instances not have the methods `getPieceList()` and `addPiece()`?